lib/publiccloud/instance.pm

```diff
@@ -273,9 +274,17 @@ sub softreboot
 
     my $duration;
 
+    # On TUNNELED test runs, ensure we are not on the publiccloud instance
+    my $prev_console = current_console();
+    # Need to store in variable, because the var can be altered
+    my $tunneled = get_var('TUNNELED', 0) && get_var("_SSH_TUNNELS_INITIALIZED", 0);
```

Comment:
I don't understand what the `&&` means here. Can you elaborate on what possible scenarios can happen over here?

Reply:
I guess to make sure that we have reached the point that the tunnel is established: https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/publiccloud/utils.pm#L44

Reply:
Then let's use `testapi::check_var()`? See the [testapi docummentation](http://open.qa/api/testapi/#_check_var).

Reply:
We only need to re-establish the ssh tunnel after rebooting, if we are in a TUNNELED test run and if the ssh tunnel has been established before.

I also updated the comment accordingly.

Reply:
That's better :smile: Thanks